src/main/java/org/jenkinsci/plugins/fodupload/SharedUploadBuildStep.java

```diff
@@ -586,4 +605,146 @@ public int getScanId() {
     public int setScanId(int newScanId) {
         return scanId = newScanId;
     }
+
+    private Path packageScanCentral(FilePath srcLocation, FilePath scanCentralLocation, FilePath outputLocation, JobModel job, PrintStream logger, Run<?, ?> build) {
+        BufferedReader stdInputVersion = null, stdInput = null;
+
+        try {
+            //version check
+            String scanCentralbatLocation = Paths.get(String.valueOf(scanCentralLocation)).resolve("scancentral.bat").toString();
+            ArrayList scanCentralVersionCommandList = new ArrayList<>();
+            scanCentralVersionCommandList.add(scanCentralbatLocation);
+            scanCentralVersionCommandList.add("--version");
+            Process pVersion = runProcessBuilder(scanCentralVersionCommandList, scanCentralLocation);
+
+            stdInputVersion = new BufferedReader(new InputStreamReader(
+                    pVersion.getInputStream()));
+            String versionLine = null;
+            String scanCentralVersion = null;
+            while ((versionLine = stdInputVersion.readLine()) != null) {
+                if (versionLine.contains("version")) {
```

Comment:
This while loop can cause issues. I'm exhausted, we'll talk about it in the morning

Reply:
Commit code as discussed.